src/main/java/uk/gov/hmcts/reform/feature/security/UserDetailsConfigurer.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.reform.feature.security;
+
+import org.springframework.security.config.annotation.authentication.configurers.provisioning.UserDetailsManagerConfigurer;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.UserDetailsManager;
+import rx.functions.Action3;
+import uk.gov.hmcts.reform.feature.webconsole.Ff4jUsersConfig;
+import uk.gov.hmcts.reform.feature.webconsole.Ff4jUsersConfig.UserDetails;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.feature.security.Roles.ROLE_ADMIN;
+import static uk.gov.hmcts.reform.feature.security.Roles.ROLE_EDITOR;
+import static uk.gov.hmcts.reform.feature.security.Roles.ROLE_USER;
+
+public class UserDetailsConfigurer {
+
+    private final UserDetailsManager manager;
+
+    private final Action3<String, String, String[]> createUser;
+
+    public UserDetailsConfigurer(UserDetailsManagerConfigurer<?, ?> configurer, PasswordEncoder passwordEncoder) {
+        this.manager = configurer.getUserDetailsService();
+        this.createUser = (String username, String password, String... roles) -> configurer
+            .withUser(username)
+            .password(passwordEncoder.encode(password))
+            .roles(roles);
+    }
+
+    public void configure(Ff4jUsersConfig userConfig) {
+        configureUsers(userConfig.getAdmins(), ROLE_ADMIN, ROLE_EDITOR);
+        configureUsers(userConfig.getEditors(), ROLE_EDITOR);
+        configureUsers(userConfig.getReaders(), ROLE_USER);
+    }
+
+    private void configureUsers(List<UserDetails> userDetails, String... roles) {
+        userDetails.forEach(user -> {
+            final String username = user.getUsername();
+            final String password = user.getPassword();
```

Comment:
do we really want those?

Reply:
sorry, you need to be more specific

Reply:
what's the benefit of adding those two constants? `user.getPassword()` is pretty clear

Reply:
First of all This is moving code around and not breaking anything in place. Second and last of all - if it is used in one place - I don't mind scrapping the local var and just using the getter directly in relevant place

Reply:
I think that code was already there but yes we can scrapping local var if it is used only once.